File: pdu/python/drone_srv_msgs/pdu_cdr_conv_DroneGoToResponse.py
from .pdu_pytype_DroneGoToResponse import DroneGoToResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DroneGoToResponse(writer: CdrWriter, src: DroneGoToResponse):
    writer.write_bool(src.ok)
    writer.write_string(src.message)


def cdr_body_to_py_DroneGoToResponse(reader: CdrReader, dst: DroneGoToResponse):
    dst.ok = reader.read_bool()
    dst.message = reader.read_string()
    return dst


def py_to_cdr_DroneGoToResponse(src: DroneGoToResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DroneGoToResponse(writer, src)
    return writer.bytes()


def cdr_to_py_DroneGoToResponse(cdr_payload) -> DroneGoToResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DroneGoToResponse()
    return cdr_body_to_py_DroneGoToResponse(reader, dst)

File: pdu/javascript/drone_srv_msgs/pdu_cdr_conv_DroneGoToResponse.js
import { DroneGoToResponse } from './pdu_jstype_DroneGoToResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDroneGoToResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DroneGoToResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_bool(src.ok);
        writer.write_string(src.message);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DroneGoToResponse } dst
     * @returns { DroneGoToResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.ok = reader.read_bool();
        dst.message = reader.read_string();
        return dst;
    }

    /**
     * @param { DroneGoToResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DroneGoToResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DroneGoToResponse());
    }
}
